File: lesson_3/code/step_by_step.m
% ----- Initiate -----
clear, close 
% Constants
    area = 100;
    population = 1000;
    infected = 10;
    iterations = 1500;
    infection_probability = 0.80;
    cure_constant = 0.01;

% Position of a person
    position = randi(area, population, 2);  

% Status - susceptible: 1, infeced 2: recovered: 3
    status = ones(population, 1); % set 1 as default
    status(1:infected)= 2;        % first are infected

    
for i = 1:iterations

    for p = 1:population
        [position(p,1),position(p,2)] = update_position(position(p,1),position(p,2), area);
    end

figure(1)
clf
hold on;
    plot(position(status == 1, 1), position(status == 1, 2),'o', 'MarkerFaceColor', 'green');
    plot(position(status == 2, 1), position(status == 2,2), 'o', 'MarkerFaceColor', 'red')
    plot(position(status == 3, 1), position(status == 3,2), 'o', 'MarkerFaceColor', 'blue')
hold off;
axis square;
end

File: lesson_3/code/step_by_step.py
import matplotlib.pyplot as plt
import numpy as np

from update_position import update_position


area = 100
population = 1000
infected = 10
iterations = 1500

position = np.random.randint(1, area + 1, size=(population, 2))
status = np.ones(population, dtype=int)
status[:infected] = 2

for _ in range(iterations):
    for person in range(population):
        x, y = update_position(int(position[person, 0]), int(position[person, 1]), area)
        position[person] = [x, y]

    plt.figure(1)
    plt.clf()
    plt.scatter(position[status == 1, 0], position[status == 1, 1], c="green", s=20)
    plt.scatter(position[status == 2, 0], position[status == 2, 1], c="red", s=20)
    plt.scatter(position[status == 3, 0], position[status == 3, 1], c="blue", s=20)
    plt.axis("square")
    plt.xlim(0, area)
    plt.ylim(0, area)
    plt.pause(0.001)

plt.show()
